main/tests/server/src/com/google/refine/RefineTest.java

```diff
@@ -431,6 +434,29 @@ protected ButterflyModule getCoreModule() {
         return coreModule;
     }
 
+    /**
+     * Runs an operation on a project. If it's a long-running operation, its process is run in the main thread until
+     * completion.
+     * 
+     * @returns the duration of the operation in milliseconds
+     */
+    protected long runOperation(AbstractOperation operation, Project project) throws Exception {
+        long start = System.currentTimeMillis();
+        Process process = operation.createProcess(project, new Properties());
+        if (process.isImmediate()) {
+            process.performImmediate();
+        } else {
+            LongRunningProcessStub longRunning = new LongRunningProcessStub((LongRunningProcess) process);
+            longRunning.run();
+        }
+        long end = System.currentTimeMillis();
+        return end - start;
+    }
+
+    /**
+     * @deprecated use {@link #runOperation(AbstractOperation, Project)}
+     */
+    @Deprecated
```

Comment:
How much would it speed things up to drop the polling interval from 200 to 100 (or 50 or lower) ms? Test efficiency is good, but it's also valuable to stay as close as possible to how the production code works.

Reply:
I think the speed up will be marginal in most cases. Same as above: if you prefer creating another thread and polling on it, we can revert to that.

Reply:
Feel free to drop the polling interval if you want.